pkg/storaged/dialogx.jsx

```diff
@@ -50,10 +50,38 @@
                      [ child, ... ])
 
    The "tag" is used to uniquely identify this field in the dialog.
-   The action function will receive the values of all field in an
+   The action function will receive the values of all fields in an
    object, and the tag of a field is the key in that object, for
    example.  The tag is also used to interact with a field from tests.
 
+   ACTION FUNCTIONS
+
+   The action funtion is called like this:
+
+      action(values, progress_callback)
+
+   The "values" parameter contains the validated values of the dialog
+   fields and the "progress_callback" can be called by the action function
+   to update the progress information in the dialog while it runs.
+
+   The progress callback should be called like this:
+
+      progress_callback(message, cancel_callback)
+
+   The "message" will be displayed in the dialog and if "cancel_callback" is
+   not null, the Cancel button in the dialog will be enabled and
+   "cancel_callback" will be called when the user clicks it.
+
+   The return value of the action function is normally a promise.  When
+   it is resolved, the dialog is closed.  When it is rejected the value
+   given in the rejection is displayed as an error in the dialog.
+
+   If the error value is a string, it is displayed as a global failure
+   message.  When it is an array, it contains errors for individual
```

Comment:
array -> object

Reply:
I made a FIXUP commit for this, to keep the evidence that all tests have been green.